balancer/rls/balancer.go

```diff
@@ -303,7 +303,10 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 		// `resizeCache` boolean) because `cacheMu` needs to be grabbed before
 		// `stateMu` if we are to hold both locks at the same time.
 		b.cacheMu.Lock()
-		b.dataCache.resize(newCfg.cacheSizeBytes)
+		evicted := b.dataCache.resize(newCfg.cacheSizeBytes)
+		if evicted {
+			b.sendNewPickerLocked()
```

Comment:
I believe the async picker update is triggered by line 293:
```go
b.updateCh.Put(resumePickerUpdates{done: done})
```
The `resumePickerUpdates` event is processed asynchronously in the `run` method which calls `b.sendNewPickerLocked()`: https://github.com/grpc/grpc-go/blob/c04b085930ce33ee83cc3f92dbe7632031e127a9/balancer/rls/balancer.go#L214-L220

Shouldn't we ensure that the update triggered in line 293 in synchronous?

Reply:
Just realised that we're already waiting for `<-done`  in line 295, so the update is synchronous. Ignore this thred.